tests/test_multi_databases.py

```diff
@@ -0,0 +1,53 @@
+from unittest import TestCase, mock
+from sqlalchemy import Column, BigInteger, String
+
+from ._common import db, User, mysql_url, mysql_url_1, mysql_url_2
+
+
+class Foo(db.Model):
+    __bind_key__ = 'foo'
+    __tablename__ = 'foo'
+
+    id = Column(BigInteger, primary_key=True)
+    foo = Column(String(64))
+
+    def __init__(self, foo):
+        self.foo = foo
+
+
+class MultiDatabasesTestCase(TestCase):
```

Comment:
Is there a way to test `__bind_key__`? Basically I'd like to make sure that if there is another model called `Bar` with `__bind_key__` set to `bar`, it's being stored in a different database.

Reply:
Sure, Let me add tests for this case

Reply:
It‘s really a problem,  Models inherit from the same Base with the same __tablename__ can not work.

It's one possible solution: https://gist.github.com/aleiphoenix/5925631

Reply:
Another solution is creating a separate Base  for each bind.

Reply:
Or just save the objects using the normal `session`, but then query the individual databases using lower-level SQLAlchemy functions. Something like following:

```python
session = db.sessionmaker()

session.add(Foo('foo'))
session.add(Bar('first bar'))
session.add(Bar('second bar'))
session.commit()

foo_count = db.get_engine('foo').execute("SELECT COUNT(*) FROM foo")
bar_count = db.get_engine('bar').execute("SELECT COUNT(*) FROM bar")

# assert foo_count is 1 and bar_count is 2
```

Reply:
I just write an example of how to solve the same-tablename problem,

https://github.com/dingyaguang117/tornado-sqlalchemy/blob/1b93049c8622267ef5cd0c78e961d3df6d81d9e7/tests/test_multi_databases.py#L7

It's not that elegant, but it works

Reply:
maybe we could improve it at next version

Reply:
Why do we need a new base? I thought that `__bind_key__` is responsible for making sure that objects end up in the correct database.

Reply:
To avoid same-tablename, you can just have two models - `Foo` (with `__bind_key__` set to `foo`), and `Bar` (with `__bind_key__` set to `bar`). No other models would be necessary. The purpose of the test case is just to ensure that objects of those models end up in their targeted databases.

Reply:
SQLAlchemy doesn't native support __bind_key__, SQLAlchemy distinguishes tables only by __tablename__.

Reply:
OK, I got you. Let me write another test case for the `__bind_key__` test.